test/e2e/nstemplatetier_test.go

```diff
@@ -0,0 +1,44 @@
+package e2e
+
+import (
+	"context"
+	"os"
+	"testing"
+
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/pkg/apis/toolchain/v1alpha1"
+	"github.com/codeready-toolchain/toolchain-e2e/doubles"
+	"github.com/codeready-toolchain/toolchain-e2e/wait"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	client "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func TestCreateNSTemplateTierAtStartup(t *testing.T) {
+	// given:
+	// given
+	tierList := &toolchainv1alpha1.NSTemplateTierList{}
+	ctx, awaitility := doubles.InitializeOperators(t, tierList)
+	defer ctx.Cleanup()
+	hostNs := os.Getenv(wait.HostNsVar)
+
+	// when
+	err := awaitility.Client.List(context.TODO(), client.InNamespace(hostNs), tierList)
```

Comment:
You can't list the objects right away. They are created/updated asynchronously. You have to wait for them. Take a look at https://github.com/codeready-toolchain/toolchain-e2e/blob/84651717e851f21de951bc30590f3dee9be1d7ba/test/e2e/e2e_test.go#L205 for example.

Reply:
ah ok, let me do that

Reply:
test refactored in b605ffe